src/travel-details-screens/use-realtime-text.ts

```diff
@@ -1,21 +1,30 @@
 import {usePreferences} from '@atb/preferences';
 import {DepartureDetailsTexts, useTranslation} from '@atb/translations';
-import {EstimatedCallWithMetadata} from '@atb/travel-details-screens/use-departure-data';
 import {formatToClock, isInThePast} from '@atb/utils/date';
 import {getTimeRepresentationType} from '@atb/travel-details-screens/utils';
 
-export const useRealtimeText = (
-  estimatedCallsWithMetadata: EstimatedCallWithMetadata[],
-) => {
-  const lastPassedStop = estimatedCallsWithMetadata
-    .filter((a) => a.actualDepartureTime)
-    .pop();
+export type EstimatedCall = {
```

Comment:
I'm not sure about the naming here and whether this is a good approach for typing what the hook should take in

Reply:
Had to look at this locally, and noticed that we can simplify this quite a bit if we set `lastPassedStop` and `quay` to optionals here, then simplify usage:

DepartureDetailsScreenComponent:
```ts
const realtimeText = useRealtimeText({
  lastPassedStop,
  firstStop,
});
```

TripSection:
```ts
const realtimeText = useRealtimeText({
  lastPassedStop,
  firstStop: leg.serviceJourneyEstimatedCalls[0],
});
```

Would also rename this to something like `RealtimeTextParams` and removed the export when it's no longer used.

Reply:
Thanks for the input @rosvik ! I had a look at this with @gorandalum, and we found a different solution that made fewer changes to the original implementation which we thought was a bit better :)